compliance_tool/cli.py

```diff
@@ -1,57 +1,67 @@
 import argparse
-from compliance_tool import list, scan, remediate
+from compliance_tool import list_, scan, remediate
 
 
-def prepare_parser_list(subparsers) -> None:
-    parser = subparsers.add_parser("list",
-                                   help="list available components")
-    parser.set_defaults(func=list.execute)
+def prepare_parser_list(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("list", parents=[global_parser],
+                                        help="list available components")
+    parser.set_defaults(func=list_.execute)
     list_subparsers = parser.add_subparsers(required=True)
 
     list_subparsers.add_parser("profiles",
                                help="list available profiles")
 
-    parser = list_subparsers.add_parser("controls",
-                                        help="list available controls for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("controls", parents=[profile_parser],
+                               help="list available controls for a given profile")
 
-    parser = list_subparsers.add_parser("rules",
-                                        help="list available rules for a given profile")
-    parser.add_argument("--profile")
+    list_subparsers.add_parser("rules", parents=[profile_parser],
+                               help="list available rules for a given profile")
 
 
-def prepare_parser_scan(subparsers) -> None:
-    parser = subparsers.add_parser("scan",
-                                   help="perform a compliance scan")
+def prepare_parser_scan(cli_sub_parsers, global_parser, profile_parser) -> None:
+    parser = cli_sub_parsers.add_parser("scan", parents=[global_parser, profile_parser],
+                                        help="perform a compliance scan")
     parser.set_defaults(func=scan.execute)
-    parser.add_argument("--profile")
-    parser.add_argument("--control")
-    parser.add_argument("--rule")
 
-    parser.add_argument("--html")
-    parser.add_argument("--json")
+    parser.add_argument("--control", metavar="CONTROL_ID",
+                        help="control identifier")
```

Comment:
The help text is very short here. It isn't clear what will happen if a user provides this argument. We need to explain in the help text what would happen if the user provides this argument.

Also, please foresee that we will use the help text to generate manual and or docs.

Reply:
Expanded it a bit. I would expect us to add more details following implementation of these features.